stdlib/2and3/socket.pyi

```diff
@@ -540,8 +540,11 @@ class socket:
     def getsockopt(self, level: int, optname: int, buflen: int) -> bytes: ...
 
     def gettimeout(self) -> Optional[float]: ...
-    def ioctl(self, control: object,
-              option: Tuple[int, int, int]) -> None: ...
+    if sys.platform == 'win32':
+        if sys.version_info >= (3, 6):
+            def ioctl(self, control: int, option: Union[int, Tuple[int, int, int], bool]) -> None: ...
```

Comment:
Would it be possible to use Literal overloads for both of these? Given that if certain values are provided, the `option` type is known exactly. See [this file](/python/typeshed/blob/c76a298ffa84c1272ddaa04f9b27eae9a011f027/stdlib/3/tempfile.pyi#L28) for an example

Reply:
I tried this earlier:

```python
def ioctl(self, control: Literal[SIO_RCVALL], ...)
```

It is not allowed, because these constants are not literals.

```
stdlib\2and3\socket.pyi:543: error: Parameter 1 of Literal[...] is invalid
stdlib\2and3\socket.pyi:543: error: Variable "socket.SIO_RCVALL" is not valid as a type
```

Reply:
It seems to me you would have to use the actual value, EG `2550136833`. I guess not great then, as that's both hard to read and unlikely to be passed literally.

Reply:
You'd probably have to define the constants as e.g. `SIO_RCVALL: Final = 123` in the stub.

Reply:
Is is possible to run mypy with a custom typeshed version (the one i cloned). I want to test some of the `Literal` combinations.

Reply:
Yes, `mypy --custom-typeshed-dir`

Reply:
@CraftSpider Based on the rest of the socket module, I think the most-appropriate solution might be to redefine the `ioctl` options as an `IntEnum` much like `AddressFamily` and or `MsgFlag`.